### PR TITLE
[android][media-library] Read video dimensions from MediaStore cursor directly

### LINKED ISSUE #44713: [expo-media-library][Android] getAssetsAsync extremely slow due to MediaMetadataRetriever opening every video file
### Minimal reproducible example

https://github.com/oeddyo/slow-expo-media-query-repro

### Steps to reproduce

  1. Clone the repo: git clone https://github.com/oeddyo/slow-expo-media-query-repro
  2. npm install && npx expo run:android (requires a development build, not Expo Go)
  3. Use a physical Android device with 50+ videos in the gallery
  4. Grant media permissions when prompted
  5. Tap "Fetch 200" and observe the timing
  6. Tap "Fetch 2000" — notice it scales linearly with the number of videos in the batch

  Platform: Android (physical device)
  Environment: development build (npx expo run:android)
  Package manager: npm

  Expected: getAssetsAsync should return in <200ms for 200 assets (as it does on iOS)
  Actual: Takes ~2s for 200 assets, ~12s for all 10,000+ assets

  Root cause: getAssetDimensionsFromCursor() in AssetUtils.kt uses MediaMetadataRetriever
  for every video, which opens each file from disk (~140ms/video). The dimensions are
  already available in the MediaStore cursor via WIDTH/HEIGHT columns.

### Environment

```text
expo-env-info 2.0.12 environment info:
      System:
        OS: macOS 26.0.1
        Shell: 5.9 - /bin/zsh
      Binaries:
        Node: 20.19.6 - ~/.nvm/versions/node/v20.19.6/bin/node
        Yarn: 1.22.19 - /opt/homebrew/bin/yarn
        npm: 10.8.2 - ~/.nvm/versions/node/v20.19.6/bin/npm
      Managers:
        CocoaPods: 1.16.2 - /opt/homebrew/bin/pod
      SDKs:
        iOS SDK:
          Platforms: DriverKit 25.0, iOS 26.0, macOS 26.0, tvOS 26.0, visionOS 26.0, watchOS 26.0
      IDEs:
        Android Studio: 2024.2 AI-242.23726.103.2422.13103373
        Xcode: 26.0.1/17A400 - /usr/bin/xcodebuild
      npmPackages:
        expo: ~54.0.33 => 54.0.33
        react: 19.1.0 => 19.1.0
        react-native: 0.81.5 => 0.81.5
      npmGlobalPackages:
        eas-cli: 16.28.0
      Expo Workflow: managed
```

### Expo Doctor Diagnostics

```text
Running 17 checks on your project...
  17/17 checks passed. No issues detected!
```

### PR DESCRIPTION
**Pull request by oeddyo**
## Summary

On Android, `getAssetDimensionsFromCursor()` uses `MediaMetadataRetriever` for every video asset, which opens each file from disk (~140ms per video). This makes `getAssetsAsync()` extremely slow on devices with many videos.

MediaStore already has `WIDTH` and `HEIGHT` columns populated by the media scanner. This PR reads dimensions from the cursor first and only falls back to `MediaMetadataRetriever` when the cursor returns 0 (rare — only for files not yet indexed by the media scanner).

## Benchmarks

Physical Android device with 10,611 assets (448 videos):

| Batch size | Before | After | Speedup |
|---|---|---|---|
| 500 assets | ~2,200ms | ~180ms | ~12x |
| 5,000 assets | ~5,500ms | 883ms | ~6x |
| 10,611 assets (all) | 11,768ms | 1,609ms | **7.3x** |

## Why this is safe

1. **Same data source.** MediaStore populates `WIDTH`/`HEIGHT` using `MediaMetadataRetriever` internally when the media scanner indexes a file — values are identical.
2. **Verified empirically.** Compared cursor dimensions vs `MediaMetadataRetriever` dimensions for all 448 videos on a physical device — zero mismatches.
3. **Graceful fallback.** The `> 0` guard ensures we still use `MediaMetadataRetriever` for any edge case where MediaStore hasn't indexed the file yet.
4. **No change to image path.** Only the video code path is affected.

## Test plan

- [x] Compared cursor and `MediaMetadataRetriever` output for 448 videos — 0 mismatches
- [x] Test on device with freshly transferred videos (to verify fallback path)

## Related issue

Fixes #44713

## Minimal reproducible example

https://github.com/oeddyo/slow-expo-media-query-repro

### CODE REVIEW

**Review by Wenszel — APPROVED** [review]
Hey! Great work on the benchmarks - the results look amazing! Just left one small note. Looking forward to merging this!

File: packages/expo-media-library/android/src/main/java/expo/modules/medialibrary/assets/AssetUtils.kt [hunk 1]
```diff
@@ -220,6 +220,16 @@ fun getAssetDimensionsFromCursor(
 ): Pair<Int, Int> {
   val uri = cursor.getString(localUriColumnIndex)
   if (mediaType == MediaStore.Files.FileColumns.MEDIA_TYPE_VIDEO) {
+    // Fast path: read dimensions from MediaStore cursor (no file I/O).
+    // MediaStore populates these when the media scanner indexes the file.
+    val cursorWidth = cursor.getInt(cursor.getColumnIndex(MediaStore.MediaColumns.WIDTH))
+    val cursorHeight = cursor.getInt(cursor.getColumnIndex(MediaStore.MediaColumns.HEIGHT))
+    if (cursorWidth > 0 && cursorHeight > 0) {
+      val orientation = cursor.getInt(cursor.getColumnIndex(MediaStore.Images.Media.ORIENTATION))
```
Comment (Wenszel) [inline packages/expo-media-library/android/src/main/java/expo/modules/medialibrary/assets/AssetUtils.kt]:
```suggestion
    val widthIndex = cursor.getColumnIndex(MediaStore.MediaColumns.WIDTH)
    val heightIndex = cursor.getColumnIndex(MediaStore.MediaColumns.HEIGHT)
    val width = cursor.getInt(widthIndex)
    val height = cursor.getInt(heightIndex)
    if (cursorWidth > 0 && cursorHeight > 0) {
      val orientationIndex = cursor.getColumnIndex(MediaStore.Images.Media.ORIENTATION)
      val orientation = cursor.getInt(orientationIndex)

```

**Comment by oeddyo** [comment]
Updated per review
IIUC only owners/reviewers can merge the change? I'll leave it to you 🙌  @Wenszel

**Comment by oeddyo** [comment]
@Wenszel I might be too greedy but is it possible to backport it to 55? We rely on this change and I want to get rid of the patch 😄

https://github.com/expo/expo/pull/44819
